dimagi/utils/logging.py

```diff
@@ -29,8 +29,11 @@ def notify_error(message):
 
 
 def notify_exception(request, message=None, details=None):
+    if request is not None:
+        message = message or request.path
     notify_logger.error(
-        'Notify Exception: %s' % (message or request.path),
+        'Notify Exception: %s' % (message
+                                  or "No message provided, fix error handler"),
```

Comment:
I looked at where notify exception gets called, https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/sms/api.py#L63, and request = None is passed to it. that line would certainly throw an error and probably issue another notify exception with the bad subject line. No?